Update docs to reflect docker_container_use_file is now true (#12529)
DataDog/documentation: content/en/logs/guide/docker-logs-collection-troubleshooting-guide.md

## Docker log collection from file issues

> Docker logs are collected from the log files on disk by default in agent 6.33.0/7.33.0+ so long as the log files on disk are accessible by the agent. `DD_LOGS_CONFIG_DOCKER_CONTAINER_USE_FILE` can be set to `false` to disable this behavior. Some containers may still be tailed from the Docker socket despite the default if the logs file path is not accessible. If the Agent ever collected logs from a given container via the docker socket, it will continue to do so (including after Agent restarts) in order to avoid sending duplicate logs. To force the Agent to collect logs from file for such a container, set `DD_LOGS_CONFIG_DOCKER_CONTAINER_FORCE_USE_FILE` to `true`. This may cause duplicate logs when the transition occurs.

Reviewer: ```suggestion
The Agent collects Docker logs from the log files on disk by default in versions 6.33.0/7.33.0+ so long as the log files on disk are accessible by the Agent. `DD_LOGS_CONFIG_DOCKER_CONTAINER_USE_FILE` can be set to `false` to disable this behavior.

The Agent tails containers from the Docker socket if it cannot access the logs file path. If the Agent ever collected logs from a particular container using the Docker socket, it will continue to do so (including after Agent restarts) in order to avoid sending duplicate logs. To force the Agent to collect logs from file, set `DD_LOGS_CONFIG_DOCKER_CONTAINER_FORCE_USE_FILE` to `true`. This setting may cause duplicate logs to appear in Datadog.
```

If logs are collected but single lines appear to be split, check that the Docker daemon is using the [JSON logging driver](#your-containers-are-not-using-the-json-logging-driver).

> Docker logs are collected from the log files on disk by default. `DD_LOGS_CONFIGy_DOCKER_CONTAINER_USE_FILE` can be set to `false` to disable this behavior. Some containers may still be tailed from the Docker socket despite the default if the logs file path is not accessible.

Reviewer 2: What you mention here should be moved to the top of the section I think.

Also, typo:

```suggestion
Docker logs are collected from the log files on disk by default. `DD_LOGS_CONFIG_DOCKER_CONTAINER_USE_FILE` can be set to `false` to disable this behavior. Some containers may still be tailed from the Docker socket despite the default if the logs file path is not accessible.
```

In addition, I think we should keep a mention here of `DD_LOGS_CONFIG_DOCKER_CONTAINER_FORCE_USE_FILE`. Something to the effect of `If the Agent ever collected logs from a given container via the docker socket, it will continue to do so (including after Agent restarts) in order to avoid sending duplicate logs. To force the Agent to collect logs from file for such a container, set DD_LOGS_CONFIG_DOCKER_CONTAINER_FORCE_USE_FILE to true. This may cause duplicate logs [...]`. WDYT?

If logs are collected but single lines appear to be split, check that the Docker daemon is using the [JSON logging driver](#your-containers-are-not-using-the-json-logging-driver).

> **Note:** When an agent is installed on the host the agent does not have permission to access `/var/lib/docker/containers` and therefore logs will be collected from the docker socket. 

Reviewer: ```suggestion
**Note:** When you install the Agent on the host, the Agent does not have permission to access `/var/lib/docker/containers`. Therefore, the Agent collects logs from the docker socket when it is installed on the host. 
```

If logs are collected but single lines appear to be split, check that the Docker daemon is using the [JSON logging driver](#your-containers-are-not-using-the-json-logging-driver).

> **Note:** When you install the Agent on the host, the Agent does not have permission to access `/var/lib/docker/containers`. Therefore, the Agent collects logs from the docker socket when it is installed on the host. 

Reviewer: ```suggestion
**Note:** When you install the Agent on the host, the Agent does not have permission to access `/var/lib/docker/containers`. Therefore, the Agent collects logs from the Docker socket when it is installed on the host. 
```
